src/unsafe-keyword.md

```diff
@@ -0,0 +1,79 @@
+# The `unsafe` keyword
+
+The `unsafe` keyword in Rust can occur in several different context: unsafe
+functions (`unsafe fn`), unsafe blocks (`unsafe {}`), unsafe traits (`unsafe
+trait`), and unsafe trait implementations (`unsafe impl`). It plays several
+different roles, depending on where it is used and whether the
+`unsafe_op_in_unsafe_fn` lint is enabled:
```

Comment:
I'm always on the fence on referring to rustc's lints in the reference since they aren't exactly part of the language. However, in this case it seems very relevant here. I think we'll need to figure out a better policy in the future.

It might help to link this for someone to learn more about the lint:

```suggestion
[`unsafe_op_in_unsafe_fn`] lint is enabled:
```

```
[`unsafe_op_in_unsafe_fn`]: ../rustc/lints/listing/allowed-by-default.html#unsafe-op-in-unsafe-fn
```

Reply:
Oh I didn't know each lint has a page like that. :)